src/modules/flow/platform/platform.c

```diff
@@ -208,4 +212,75 @@ platform_machine_id_open(struct sol_flow_node *node,
         SOL_FLOW_NODE_TYPE_PLATFORM_MACHINE_ID__OUT__OUT, id);
 }
 
+static int
+hostname_send(struct sol_flow_node *node)
+{
+    int r;
+    char *name;
+
+    r = sol_plataform_get_hostname(&name);
+    SOL_INT_CHECK(r, < 0, r);
+    return sol_flow_send_string_packet(node, 0, name);
```

Comment:
see, here you're leaking! There is a "take" variant that will own `name` and free it, this one will `strdup()` and leak :-)